test/loader.spec.js

```diff
@@ -54,6 +55,22 @@ describe('srcset-loader', () => {
 		expect(artifacts.length).toBe(8);
 	});
 
+	it('should use custom generator and image url', async () => {
```

Comment:
let's also add tests for exceptions 

1. case where no publicPath
2. case where publicPath is not full url

Reply:
done